### PR TITLE
[Doc] Add doc for usage stats collection

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Add doc for usage stats collection

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/usage-stats.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+.. _ref-usage-stats
```
Comment (jjyao) [inline doc/source/cluster/usage-stats.rst:1]: Not sure if it's the right place to have it.

File: doc/source/cluster/usage-stats.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+If usage stats collection is enabled, a background process on the head node will collect the usage stats
+and report to ``https://usage-stats.ray.io/`` every hour. The reported usage stats will also be saved to
+``/tmp/ray/session_xxx/usage_stats.json`` on the head node for inspection. You can check the existence of this file to see if collection is enabled.
+
+Usage stats collection is very lightweight and should have no impact on your workload in any way.
+
+What data is collected
+----------------------
+We collect non-sensitive data that helps us understand how Ray is used (e.g. which Ray library is used).
+The personally identifiable data is never collected. Please check `usage_lib.py <https://github.com/ray-project/ray/blob/master/python/ray/_private/usage/usage_lib.py>`_ for all the data we collect.
```
Comment (rkooo567) [inline doc/source/cluster/usage-stats.rst:32]: Maybe we should mention t olook `UsageStatsToReport `?

File: doc/source/cluster/usage-stats.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Note: usage stats collection is not enabled for local dev cluster started via ``ray.init()``.
+
+If usage stats collection is enabled, a background process on the head node will collect the usage stats
+and report to ``https://usage-stats.ray.io/`` every hour. The reported usage stats will also be saved to
+``/tmp/ray/session_xxx/usage_stats.json`` on the head node for inspection. You can check the existence of this file to see if collection is enabled.
+
+Usage stats collection is very lightweight and should have no impact on your workload in any way.
+
+What data is collected
```
Comment (rkooo567) [inline doc/source/cluster/usage-stats.rst:29]: Why don't we put this on the top and "how does it work" should be at the bottom.

Comment (rkooo567) [reply doc/source/cluster/usage-stats.rst:29]: My assumption is users care things in this order
1. what data is collected
2. How to disable it
3. How it works

Comment (jjyao) [reply doc/source/cluster/usage-stats.rst:29]: I feel `How it works` makes understanding `How to disable it` easier.

**Comment by ericl** [comment]
I think @richardliaw and @dynamicwebpaige should be the approvers for this.

File: doc/source/cluster/usage-stats.rst (hunk 3 above)
Comment (dynamicwebpaige) [reply doc/source/cluster/usage-stats.rst]: I agree with @rkooo567's comment. We could minimally add a link to the  `How to disable it?` header in the introductory paragraph, similar to:

> Starting in Ray 1.13, Ray will collect API utilization statistics by default (guarded by an opt-out prompt, [see below]()). These data will be used by the open-source Ray engineering team to better understand how to improve our libraries and core APIs, and how to prioritize bug fixes and enhancements. For more context, please refer to this `RFC <https://github.com/ray-project/ray/issues/20857>`_.

cc: @richardliaw @ericl

**Review by dynamicwebpaige — APPROVED** [review]

File: doc/source/cluster/usage-stats.rst [hunk 4]
```diff
@@ -0,0 +1,52 @@
+.. _ref-usage-stats:
+
+Usage Stats Collection
+======================
+
+Starting in Ray 1.13, Ray collects usage stats data by default (guarded by an opt-out prompt).
+This data will be used by the open-source Ray engineering team to better understand how to improve our libraries and core APIs, and how to prioritize bug fixes and enhancements.
```
Comment (richardliaw) [inline doc/source/cluster/usage-stats.rst:7]:
```suggestion
This data will be used by the open-source Ray engineering team to better understand how to improve our libraries and core APIs, and how to prioritize bug fixes and enhancements.

You will always be able to `disable the usage collection <usage-disable>`_.
```

File: doc/source/cluster/usage-stats.rst [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+For more context, please refer to this `RFC <https://github.com/ray-project/ray/issues/20857>`_.
+
+What data is collected?
+-----------------------
+We collect non-sensitive data that helps us understand how Ray is used (e.g., which Ray libraries are used).
+Personally identifiable data is never collected. Please check ``UsageStatsToReport`` in `usage_lib.py <https://github.com/ray-project/ray/blob/master/python/ray/_private/usage/usage_lib.py>`_ to see the data we collect.
+
+How to disable it
```
Comment (richardliaw) [inline doc/source/cluster/usage-stats.rst:15]:
```suggestion
.. _usage-disable:

How to disable it
```

File: doc/source/cluster/usage-stats.rst [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+-----------------
+There are multiple ways to disable usage stats collection before starting a cluster:
+
+#. Add ``--disable-usage-stats`` option to the command that starts the Ray cluster (e.g., ``ray start --head --disable-usage-stats``).
```
Comment (richardliaw) [inline doc/source/cluster/usage-stats.rst:19]: Can we link to `ray start` docstring?

File: doc/source/cluster/usage-stats.rst [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+#. Run ``ray disable-usage-stats`` to disable collection for all future clusters. This won't affect currently running clusters. Under the hood, this command writes ``{"usage_stats": true}`` to the global config file ``~/.ray/config.json``.
```
Comment (richardliaw) [inline doc/source/cluster/usage-stats.rst:21]: Can we link to `ray disable-usage-stats` docstring?

File: doc/source/cluster/usage-stats.rst [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+======================
+
+Starting in Ray 1.13, Ray collects usage stats data by default (guarded by an opt-out prompt).
+This data will be used by the open-source Ray engineering team to better understand how to improve our libraries and core APIs, and how to prioritize bug fixes and enhancements.
+For more context, please refer to this `RFC <https://github.com/ray-project/ray/issues/20857>`_.
+
+What data is collected?
+-----------------------
+We collect non-sensitive data that helps us understand how Ray is used (e.g., which Ray libraries are used).
+Personally identifiable data is never collected. Please check ``UsageStatsToReport`` in `usage_lib.py <https://github.com/ray-project/ray/blob/master/python/ray/_private/usage/usage_lib.py>`_ to see the data we collect.
```
Comment (richardliaw) [inline doc/source/cluster/usage-stats.rst]: Can you document this (render it on the docs) and then link to it, instead of linking to a file here?

**Review by richardliaw — APPROVED** [review]
LGTM!

**Comment by jjyao** [comment]
Merge this without waiting for mac tests since it's a doc only PR.

**Pull request merged**